extensions/sql-migration/src/api/azure.ts

```diff
@@ -56,6 +56,18 @@ export async function getResourceGroups(account: azdata.Account, subscription: S
 	return result.resourceGroups;
 }
 
+export async function getResourceGroupByName(account: azdata.Account, subscription: Subscription, resourceGroupName: string): Promise<azureResource.AzureResourceResourceGroup> {
+	const api = await getAzureCoreAPI();
+	const path = encodeURI(`/subscriptions/${subscription.id}/resourcegroups/${resourceGroupName}?api-version=2021-04-01`);
```

Comment:
This API call might not work since this is exactly what we were trying to avoid - making an API call to look up a particular resource group. I only added this function because internally, resource groups are not stored as a list of strings of resource group names, but rather a list of resource group objects (which contains properties such as name and ID). However, I should be able to craft that manually without needing an API call. Will investigate

Reply:
No longer need an API call to construct the resource group object, now it is crafted manually based on the properties available to the individual resources.